flepimop/gempyor_pkg/src/gempyor/_pydantic_ext.py

```diff
@@ -8,16 +8,20 @@
 __all__ = ()
 
 
+import numbers
 from pathlib import Path
-from typing import Any, TypeVar, overload
+from functools import partial
+from typing import Any, TypeVar, overload, Annotated, Type
 
 import pyarrow.parquet as pq
 import pandas as pd
-from pydantic import BaseModel, RootModel
+from pydantic import BaseModel, RootModel, BeforeValidator
+import sympy
```

Comment:
```suggestion
from sympy.parsing.sympy_parser import parse_expr
```
I think we only need this one thing from a specific module rather then the entire `sympy` package?